### ISSUE TITLE
Tide-search Xcorr values seem wildly different from Thermo "PSMS" software values

### ISSUE THREAD

**Issue by wsnoble** [crux-toolkit/crux-toolkit#624]
From Michael Pique via the crux-users mailing list:

I am running Tide-search on ms2 files which our collaborator has run Thermo's PSMS software on.  The Thermo Xcorr values are like we have seen for many years, between .5 and 2 or 3.  But the Tide values are much smaller (0.6 max) and about half of them are negative.  A scatter plot show no correlation. I feel I am missing some setting but have tried  some without success.  I will attach the parameter file and a screenshot of the scattergram and the beginning of the tables.  Here is the log:

NFO: CPU: Luna.local

INFO: Crux version: 4.1-4ff75ee-2022-09-28

INFO: Wed Jan 18 12:41:40 PST 2023

COMMAND LINE: crux tide-search --use-tailor-calibration T --output-dir tide9-Elite202212a74-OPSD Elite202212a74.ms2 OPSD

INFO: Beginning tide-search.

INFO: Running tide-search...

INFO: Number of Threads: 1

INFO: Reading index OPSD

INFO: Read 1 target proteins

INFO: Reading auxiliary locations.

INFO: Reading peptide_header.

INFO: Converting Elite202212a74.ms2 to spectrumrecords format

INFO: Elapsed time starting conversion: 0.00688 s

INFO: Converting ms_level 2 ... 

INFO: Reading spectrum file tide9-Elite202212a74-OPSD/Elite202212a74.ms2.spectrumrecords.tmp.

INFO: Read 5505 spectra.

INFO: Starting search.

INFO: 1000 spectrum-charge combinations searched, 18% complete

INFO: 2000 spectrum-charge combinations searched, 36% complete

INFO: 3000 spectrum-charge combinations searched, 54% complete

INFO: 4000 spectrum-charge combinations searched, 73% complete

INFO: 5000 spectrum-charge combinations searched, 91% complete

INFO: [Thread 0]: Deleted 0 precursor, 0 isotope and 0 out-of-range peaks.

INFO: [Thread 0]: Retained 100% of peaks.

INFO: Time per spectrum-charge combination: 0.000476 s.

INFO: Average number of candidates per spectrum-charge combination: 1.388193 

INFO: Elapsed time: 2.62 s

INFO: Finished crux tide-search.

INFO: Return Code:0
![Scattergram_Tide_Xcorr_vs_PSMS_Xcorr](https://user-images.githubusercontent.com/22529318/213535591-ad313c15-b18c-415d-aaf4-df27ba920a79.png)

**Comment by wsnoble** [comment]
This is indeed strange since we expect to see some correlation. However, I noticed that you are not actually plotting the Tide XCorr -- you are plotting the Tide Tailor score. Can you set `--use-tailor-calibration F` and re-do the plot?

**Comment by wsnoble** [comment]
Michael Pique wrote: 

Thanks for looking at this so fast.  I sent Andy Lin a description of what I did to extract the sequences and the Xcorrs from the PSMS and Tide output files.

I often make mistakes - but looking again at the "tide-search.target.txt", I'm pulling out and plotting the ninth tab-separated field, "xcorr score" not the tenth, "tailor score". I just reran with "--use-tailor-calibration F" and get precisely the same Xcorr values, only difference I see is that the "tailor score" header and data field are gone.  I can pull out the first hundred or so lines of each and send if helpful.

**Comment by wsnoble** [comment]
OK, thanks for clarifying this.  It would be good to pull out the fields that list the scan number, the peptide sequence, and the XCorr score from both the Tide output and the PSMS output, and put those together into a six-column file.  Then we can verify that the two XCorr scores are truly coming from the same PSM.  Is this what you've done, and if so, can you send that file?

**Comment by mpiqueca** [comment]
I will do that right away (almost, have to run a quick errand) for the peptides that are in common between the two.   I’ve not double-checked that Tide doesn’t repeat pep-ions in the output; that will make the merge a little trickier.  Let me add the scan numbers from each.  The sequence I already check is the same (that, and the ion/charge, is how I detect the common peptides.

**Comment by mpiqueca** [comment]
Here (attached) is the TSV file including the scan numbers.  Most are the same - you will see some are different. I suspect this is because I currently pull out from Tide only the top-ranked PSM (xcorr rank 1).   I can generalize this but wanted to send this out before I stop for now.  Big thanks for your suggestions and patience.

[202212a74_xcorrmin-1_psms_tide.tsv.txt](https://github.com/crux-toolkit/crux-toolkit/files/10461741/202212a74_xcorrmin-1_psms_tide.tsv.txt)

**Comment by wsnoble** [comment]
Thanks.  Is the column labeled "Sequence" from Tide or from PSMS?  Can you extract the peptide sequence from both methods? What is the meaning of "pepid"?

Also, what version of Crux are you using? You can find this in the first few lines of the tide-search.log.txt file.

I agree that, qualitatively, the Tide XCorr scores seem weird!

**Comment by bmx8177** [comment]
The Crux version he is using is below (took from screen shot in original post)

INFO: Crux version: 4.1-4ff75ee-2022-09-28

**Comment by kfattila** [comment]
It looks like the tide is not parameterized well. Maybe same static mod or TMT labeling is missing.It would be nice to see all fields of the PSMs. Maybe the joining of the two results files from Tide and Thermo isn't correct.

**Comment by mpiqueca** [comment]
The “Sequence” column is (by construction) the same for both Thermo “PSMS” and Crux - from that sequence, I find its beginning and end residue numbers by looking up the sequence string in the FASTA file (checking that it is unique within that FASTA protein sequence).   The “pepid” is the beginning and end of the peptide, with its charge.  It is the key on which the Thermo and Crux output files are joined.   Currently, my joining function allows for only one occurrence of a given key (pepid) in each input file.  I can later generalize that to add a “scan number” to the join key, but that’s a bit of a distraction from figuring out my problem with the Xcorr values.     As to all fields of the Thermo and Tide files, in an email yesterday to Andy Lin, I sent the matching lines from each file for one of the peptides, along with the header line that identifies the fields.  I can post that here if helpful.  Thank you all for your suggestions.

**Comment by wsnoble** [comment]
Yes, please post here.
Thanks!

**Comment by wsnoble** [comment]
It would also be helpful to see the Tide log file.

**Comment by mpiqueca** [comment]
Here are minimal output files from both the Thermo PSMS and Tide-search - a header line and one peptide detection (20-29+2 at scan number 3015).  Thermo has Xcorr 2.05, Tide has 0.21. I also made a stripped-down MS2 file that has only that scan 3015. (Like the .tsv files, I had to add a ".txt" for Girhub to allow uploading it.)

[thermo_onepeptide.tsv.txt](https://github.com/crux-toolkit/crux-toolkit/files/10469268/thermo_onepeptide.tsv.txt)
[tide_onepeptide.tsv.txt](https://github.com/crux-toolkit/crux-toolkit/files/10469281/tide_onepeptide.tsv.txt)
[tide-search.log.txt](https://github.com/crux-toolkit/crux-toolkit/files/10469291/tide-search.log.txt)
[tide-search.params.txt](https://github.com/crux-toolkit/crux-toolkit/files/10469293/tide-search.params.txt)
[Elite202212a74_onescan3015.ms2.txt](https://github.com/crux-toolkit/crux-toolkit/files/10469360/Elite202212a74_onescan3015.ms2.txt)

**Comment by wsnoble** [comment]
Thanks for sending these.  I can't see what the source of the discrepancy is.  I did verify that, using the current version of Tide, the XCorr scores in Comet and Tide are very concordant:

![image](https://user-images.githubusercontent.com/22529318/213787056-83df266d-f690-446a-a641-ab58893d3f30.png)

One thing I noticed is that you are using a version of Tide from September, which predates the recent major overhaul described in [this paper](https://pubs.acs.org/doi/10.1021/acs.jproteome.2c00617). Thus, I think it might be worthwhile for you to download the latest version of Tide and re-run this test.

You may also want to try running Comet on this same data.

**Comment by mpiqueca** [comment]
Thanks for continuing suggestions.   I downloaded Crux version 4.1-f575777-2023-01-13 and tried it.  The Xcorr values are exactly the same as before.  I checked this by editing the "S_0.0000" in the newer Crux output to "S_0.00" and the output files are byte-for-byte identical.   I will give Comet another try; it had been my understanding that Tide would give results closer to Thermo PSMS since it is a SEQUEST descendent. Our workflow for screening peptides is based on Xcorr values developed in the Woods/Sheng UCSD lab so we are hoping to feed that pipeline from Crux and not have to ask USCD's Sheng to do all the MS2 interpretations.

**Comment by wsnoble** [comment]
This is very strange. Can you send an ms2 file containing just scan number 3015?

FWIW, both Comet and Tide are SEQUEST descendants. :)

**Comment by mpiqueca** [comment]
I put that MS2 file in my message 2 hours ago, attached  is the OPSD FASTA file, here is the script you sent me last November that makes the OPSD index and then runs Tide.  (I again tried with the "use-tailor-calibration F" and the Xcorrs are the same as without it.    I seem to have to rename all these as ".txt" to get them through GitHub.

[OPSD.fasta.txt](https://github.com/crux-toolkit/crux-toolkit/files/10470427/OPSD.fasta.txt)
[runtide9-Elite202212a74.txt](https://github.com/crux-toolkit/crux-toolkit/files/10470430/runtide9-Elite202212a74.txt)

**Comment by wsnoble** [comment]
Is this low-resolution MS2 data? If I change the mz-bin-with from the default to ~1 Da, then the XCorr score changes from 0.214277720 to 1.87913083.

Also, can you tell me more about this "PSMS" software?  I've never heard of it.

**Comment by mpiqueca** [comment]
I’ve not thought of it as low-res, but I’m a software guy, not a spectrometrist (is that a word?).  The spacing of the m/z samples is pretty fine (see the MS2 file) as I understand it.  I can try varying the mz-bin-width tomorrow, good discovery.  Is that in Comet as well as Tide?

I don’t know the real name of the software Sheng uses. His machine is a Thermo-Fisher Orbitrap.   The peptide output files he sends us have “psms” in the name, so we’ve just called them “psms files”.  A screenshot I made of our last zoom session with him (yesterday) shows “Thermo Proteome Discoverer 1.3.0.339”.  The coverage maps he shows us are labeled “HDExaminer”.   I’ll email him and ask, should have done that earlier.

**Comment by wsnoble** [comment]
I haven't tried it in comet, but I imagine the effect would be the same.  The corresponding parameter name in Comet is [fragment_bin_tol](https://uwpr.github.io/Comet/parameters/parameters_202201/fragment_bin_tol.html).

OK, ProteomeDiscoverer is more familiar!

**Comment by wsnoble** [comment]
Michael wrote:"I'm sure I should further adjust some Comet parameters, but this looks much better.  The bottom stuff is almost certainly that my current comet-selection script just takes the rest detection, does not look for best Xcorr when a peptide is found more than once.  I got lost last year on the peptide-centric versus spectrum-centric search and never got very far."

![image](https://user-images.githubusercontent.com/22529318/213881783-381f10ae-b4a0-4be1-8867-dc69af454e9f.png)

**Comment by wsnoble** [comment]
OK, given that Comet and Tide give almost identical XCorr scores, the difference must be in how you are setting parameters. If you can tell me how you changed your parameters to get Comet to match ProteomeDiscoverer, then I can probably tell you how to make Tide do the same.

**Comment by mpiqueca** [comment]
Thanks - real progress now.  I made two changes to the Tide parameters - 
    mz-bin-width 1.0005079    (this was sufficient for Tide and Comet to have the same Xcorr, as it is Comet's default)
    peptide-centric-search T  (this was the key to let me select "xcorr rank equals 1" in my processing)

I found it best (so far) to select strictly on Xcorr rank, without considering delta_cn, but I've not dug deep into that.
(I see more scattergrams in my future).

I now have something FAR more reasonable, taking the Tide detection with highest Xcorr per peptide, I'll attach the current encouraging scattergram and tsp file.  I'll check on the outliers later.   Many thanks for the suggestions.
![Scattergram_Tide11_vs_Thermo_PSMS_Xcorr_closeup](https://user-images.githubusercontent.com/42452925/213887412-56897e53-6743-47c6-b8a1-d72b8cce8ca0.png)
[202212a74_xcorrmin0.0_psms_tide11.tsv.txt](https://github.com/crux-toolkit/crux-toolkit/files/10473099/202212a74_xcorrmin0.0_psms_tide11.tsv.txt)

**Comment by wsnoble** [comment]
Glad it's making more sense.  Note that peptide-centric=T|F should not change the score assigned to a given PSM; just which PSMs get reported.

In general, I think you should use the Tailor score rather than XCorr.  If you want to take into account delta_cn, then just use Percolator.

**Comment by mpiqueca** [comment]
Keep the suggestions coming.  I will make a Tailor Score scattergram vs Xcorr next then to see what it does.  
However, when I combined "Tailor Score T" with peptide-centric-search, there were no Tailor Scores in the output, only an orphaned header entry (first line of the tide-search.target.txt). So I turned it off to avoid having to edit that header entry out of the file.  I was going to (and will) file this as a separate trouble ticket so as not to clog this one up.

Thanks for the pointer to your new Tide paper.  My colleague here who's good at that,  has requested the full-text from the interlibrary system Scripps Research is affiliated with.
